packages/react-components/react-motion/library/src/types.ts

```diff
@@ -1,4 +1,15 @@
-export type AtomMotion = { keyframes: Keyframe[] } & KeyframeEffectOptions;
+export type AtomMotion = { keyframes: Keyframe[] } & KeyframeEffectOptions & {
+    /**
+     * Allows to specify a reduced motion version of the animation. If provided, the settings will be used when the
+     * user has enabled the reduced motion setting in the operating system. By default, the duration of the animation is
+     * set to 1ms.
+     *
+     * Note, if keyframes are provided, they will be used instead of the regular keyframes.
+     *
+     * @see https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion
+     */
+    reducedMotion?: { keyframes?: Keyframe[] } & KeyframeEffectOptions;
```

Comment:
it looks to me that `reducedMotion` is a redeclaration of `AtomMotion` 🤔, maybe it would be better to add it somewhere else to avoid repetition? or perhaps we should have a middle type here between `PresenceMotion` and `AtomMotion`?

Reply:
Maybe something similar to [griffel fallback properties](https://griffel.js.org/react/api/make-styles#css-fallback-properties)?!

```tsx
const Motion = createPresenceComponent([
  // default motion
  {
    enter: {
      keyframes: [
        { opacity: 0, transform: 'scale(0)' },
        { opacity: 1, transform: 'scale(1)' },
      ],
      duration: 300,
    },
    exit: {
      /* ... */
    },
  }, 
  {/* reduced motion goes here */}
]);
```

Reply:
@bsunderhus I also think a middle type would be good, to DRY up 
`{ keyframes: Keyframe[] } & KeyframeEffectOptions`.

Reply:
> _from @bsunderhus_ 
> or perhaps we should have a middle type here between `PresenceMotion` and `AtomMotion`?

> _from @robertpenner_ 
> I also think a middle type would be good, to DRY up
> `{ keyframes: Keyframe[] } & KeyframeEffectOptions`.

Do you mean a new type to extract `{ keyframes?: Keyframe[] } & KeyframeEffectOptions`? 

----

> Maybe something similar to [griffel fallback properties](https://griffel.js.org/react/api/make-styles#css-fallback-properties)?!

@bsunderhus I updated the PR description with two options and added some arguments against them. Please check and let me know WDYT.

Reply:
@layershifter 

> Do you mean a new type to extract { keyframes?: Keyframe[] } & KeyframeEffectOptions?

Is it necessary to make `keyframes` optional in `reducedMotion` (could we look at a use case)? Or would it be simpler to say, "If you're going to make a custom reduced motion, you need to define keyframes"?

Reply:
> Is it necessary to make keyframes optional in reducedMotion (could we look at a use case)? Or would it be simpler to say, "If you're going to make a custom reduced motion, you need to define keyframes"?

I don't think that we need to force consumers there, especially after the duration change (https://github.com/microsoft/fluentui/pull/33353#discussion_r1877844281). Also, the media query itself (https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) does not put any restrictions.